libsignal-service/src/sender.rs

```diff
@@ -538,6 +541,15 @@ where
                             .await?;
                     }
                 },
+                Err(ServiceError::ProofRequiredError(ref p)) => {
+                    log::debug!("{:?}", p);
+                    if p.options.contains(&String::from("recaptcha")) {
+                        return Err(MessageSenderError::ProofRequired {
+                            token: p.token.clone(),
+                            options: p.options.clone(),
+                        });
+                    }
```

Comment:
What about the `else` case, then it will just return an `Ok` and silently fail?

Reply:
Fixed by not filtering the captcha type here; let the client do it instead.

Reply:
Sounds good to me, we can always refine that later.